### PR TITLE
Requestapi

### PR DESCRIPTION
**Pull request by MadhuriUttarkar**

### CODE REVIEW

File: src/main/java/org/madhuri/app/message/handler/StockHandler.java [hunk 1]
```diff
@@ -31,5 +37,29 @@ public Response getStockInfo(@PathParam("symbol") String symbol) {
...
+        try {
+            if (query == null || query.trim().isEmpty()) {
+                return Response.status(Response.Status.BAD_REQUEST)
+                               .entity("Query parameter 'q' cannot be empty.")
+                               .build();
+            }
+
+            StockResponse stockResponse = stockService.searchSymbols(query);
+
+            if (stockResponse != null && stockResponse.getResult() != null && !stockResponse.getResult().isEmpty()) {
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/handler/StockHandler.java]: Why are you doing getResults twice here ? Can we not assign a variable and nil check and use same for assignment ?

File: src/main/java/org/madhuri/app/message/model/StockResponse.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+public class StockResponse {
+
+	@JsonbProperty("count")
+	private long count;
+	
+	@JsonbProperty("result")
+	private List<StockSymbol> result;
+	
+	public StockResponse()
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/StockResponse.java:16]: Do we need an empty constructor ?

File: src/main/java/org/madhuri/app/message/model/StockSymbol.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	@JsonbProperty("displaySymbol")
+	private String displaySymbol;
+	
+	@JsonbProperty("symbol")
+	private String symbol;
+	
+	@JsonbProperty("type")
+	private String type;
+
+	public StockSymbol()
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/StockSymbol.java:20]: Do we need a empty constructor ?

File: src/main/java/org/madhuri/app/message/model/StockResponse.java (hunk 2 above)
Comment (MadhuriUttarkar) [reply src/main/java/org/madhuri/app/message/model/StockResponse.java:16]: The default constructor is especially used during deserialization when converting JSON or data formats into Java objects.

File: src/main/java/org/madhuri/app/message/model/StockSymbol.java (hunk 3 above)
Comment (MadhuriUttarkar) [reply src/main/java/org/madhuri/app/message/model/StockSymbol.java:20]: The default constructor is especially used during deserialization when converting JSON or data formats into Java objects.

**Review by dwarakauttarkar — APPROVED** [review]

**Pull request merged**